calico_cni/policy_drivers.py

```diff
@@ -320,6 +307,21 @@ def _escape_chars(self, unescaped_string):
         return re.sub('[^a-zA-Z0-9\.\_\-]', swap_char, unescaped_string)
 
 
+class DefaultDenyInboundDriver(KubernetesAnnotationDriver):
+    """
+    This driver sets the labels and correct profile on the endpoint.
+    """
+    def apply_profile(self, endpoint):
+        profile_id = "k8s_ns.%s" % self.namespace
+        _log.debug("Constructed profile ID - %s" % profile_id)
+        pod = self._get_api_pod()
+        labels = pod["metadata"].get("labels", {})
```

Comment:
The labels that need to be applied are a combination of:
1) The labels from the Kubernetes pod (which you're doing here)
2) A special namespace label (which you're not).  See the policy agent code here:  https://github.com/projectcalico/k8s-policy/blob/master/handlers/pod.py#L26